README.md

```diff
@@ -1,112 +1,240 @@
+| log\_sink\_resource\_name | The resource name of the log sink that was created. |
+| parent\_resource\_id | The ID of the GCP resource in which you create the log sink. |
+| writer\_identity | The service account that logging uses to write log entries to the destination. |
+
+[^]: (autogen_docs_end)
+
+## Requirements
+### Terraform plugins
+- [Terraform](https://www.terraform.io/downloads.html) 0.11.x
+- [terraform-provider-google](https://github.com/terraform-providers/terraform-provider-google) plugin ~> v2.0.x
+
+### Configure a Service Account
+In order to execute this module you must have a Service Account with the following:
+
+#### Roles
+The service account should have the following roles:
+- `roles/logging.configWriter` on the logsink's project, folder, or organization (to create the logsink)
+- `roles/resourcemanager.projectIamAdmin` on the destination project (to grant write permissions for logsink service account)
+- `roles/serviceusage.serviceUsageAdmin` on the destination project (to enable destination APIs)
+
+#### Pub/Sub roles
+To use a Google Cloud Pub/Sub topic as the destination:
+- `roles/pubsub.admin` on the destination project (to create a pub/sub topic)
+
+To integrate the logsink with Splunk, you'll need a topic subscriber (service account):
+- `roles/iam.serviceAccountAdmin` on the destination project (to create a service account for the logsink subscriber)
+
+#### Storage role
+To use a Google Cloud Storage bucket as the destination:
+- `roles/storage.admin` on the destination project (to create a storage bucket)
+
+#### BigQuery role
+To use a BigQuery dataset as the destination, one must grant:
+- `roles/bigquery.dataEditor` on the destination project (to create a BigQuery dataset)
+
+
+### Enable API's
+In order to operate with the Service Account you must activate the following API's on the base project where the Service Account was created:
+
+- Cloud Resource Manager API - cloudresourcemanager.googleapis.com
+- Cloud Billing API - cloudbilling.googleapis.com
+- Identity and Access Management API - iam.googleapis.com
+- Service Usage API - serviceusage.googleapis.com
+- Stackdriver Logging API - logging.googleapis.com
+- Cloud Storage JSON API - storage-api.googleapis.com
+- BigQuery API - bigquery-json.googleapis.com
+- Cloud Pub/Sub API - pubsub.googleapis.com
+
+## Install
+
+### Terraform
+Be sure you have the correct Terraform version (0.11.x), you can choose the binary here:
+- https://releases.hashicorp.com/terraform/
+
+Then perform the following commands:
+
+- `terraform init` to get the plugins
+- `terraform plan` to see the infrastructure plan
+- `terraform apply` to apply the infrastructure build
+- `terraform destroy` to destroy the built infrastructure
+
+#### Variables
+Please refer the `variables.tf` file for the required and optional variables.
+
+#### Outputs
+Please refer the `outputs.tf` file for the outputs that you can get with the `terraform output` command
+
+## File structure
```

Comment:
Let's drop this section. It doesn't provide much value and easily becomes obsolete.